### ISSUE TITLE
How to make editable multilingual website pages?

### ISSUE THREAD

**Issue by mpetricek** [bleutzinn/grav-plugin-editable#9]
Hi,

I can't make editable plugin working on multilingual website. Is there way how to make it?

Thank you

**Comment by bleutzinn** [comment]
When the SimpleMDE add-on is used and the page filesystem filename contains the language it always  sets that part to "en". That is a bug.
The add-on ContentTools works okay as far as I can tell. Can you confirm that?

Can you provide some more information about how you have set up your multi language site? That may help in fixing this bug.

**Comment by mpetricek** [comment]
I'm using ContentTools but not sure how to point shortcode to the right file. For now it's showing only shortcodes on my website.

I did set up multilingual website later in the process because I didn't know that my client would need it. I'm using language switcher plugin and I did translate all the pages. Do I need to have different folders for pages or something like that?

**Comment by bleutzinn** [comment]
Do I understand correctly that the shortcode is not processed but included in the page the end-user sees just as you entered it in the backend. So for instance like:
`[editable name="introduction" /]`

Editable should use the appropriate language version of a page (e.g. `default.md`, default.es.md`, default.de.md`, etc.). You should follow the Grav documentation on multi language, not change things to try to make Editable work.

Can you post or send me (email is in the blueprints.yaml file) your system config, plus site config and some sample pages?

**Comment by mpetricek** [comment]
I've just sent you email with files which you wanted. Consider "bio-text" as test modular page.
Thank you

**Comment by bleutzinn** [comment]
Before diving into the world of multi language please try one thing.

You say the shortcodes are not processed or parsed at all. When looking at your page content I see you are using the shortcodes in HTML.
Together that makes sense. By default markdown inside HTML isn't parsed by Grav. You can tell Grav to parse it by including 'markdown="1"' in the div in which the shortcode is embedded.

Could you give that a try?

**Comment by mpetricek** [comment]
Ok so I tried. It didn't help.

You can check it out at this site
[](http://18339.w39.wedos.ws/domains/fourway.cz/)

**Comment by mpetricek** [comment]
Ok, so I made a copy of the original website and tried to turn on multilanguage functionality step by step.

Problem is obviously directly connected to "Langauge Switcher" plugin.
Straight after installation Editable stopped working.

**Comment by bleutzinn** [comment]
There was a problem that prevented the ContentTools add-on to save to the correct language page. This is fixed in [version 0.1.2](https://github.com/bleutzinn/editable-contenttools-add-on/releases/tag/v0.1.2) of the ContentTools add-on.

I've successfully set up the Multilanguage Skeleton, which demonstrates the LangSwitcher Plugin, with a working Editable Plugin and using the (0.1.2) ContentTools add-on. It works in both editable modes, that is Editable Self and Editable Regions, but these modes can not be enabled together. It is one mode or the other. To enable editable regions set `editable_self: false` in the config file.

Further tests reveal that shortcodes should not be inside HTML elements. The reason is that Editable gets the (HTML) content from the page the shortcode refers to, wraps it in a `div` element and then replaces the shortcode with the wrapped content. Chances are that this results in an invalid HTML structure.

As an example when using a shortcode in markdown:
```
## Intro

[editable name="/blog/welcome-to-grav" /]

Some text after the shortcode.
```

And when using HTML:
```
<h2>Intro</h2>
[editable name="/blog/welcome-to-grav" /]
<p>
Some text after the shortcode.
</p>
```

**Comment by mpetricek** [comment]
Hi,

so I was very happy that you found solution. Unfortunately after I moved website from localhost to my webhost the problem appears again. Again if I uninstall Langswitcher plugin everything works ok.

Curently the website is on the subdomain [here](http://fourway.petriczech.com/)
